src/Web/Intercom/Types.hs

```diff
@@ -50,23 +52,44 @@ ping c = do
 opts c = defaults & auth ?~ basicAuth (c ^. appId) (c ^. apiKey)
                   & header "Accept" .~ ["application/json"]
 
+postOpts c = opts c & header "Content-Type" .~ ["application/json"]
+
 data User = User {
   _name             :: Maybe Text,
   _email            :: Maybe Text,
   _userId           :: Maybe Text,
-  _customAttributes :: Map String Value
+  _customAttributes :: HashMap Text Value
 } deriving (Eq, Show)
 
+-- | A blank user, for overriding
+-- > let myUser = blankUser & email .~ (Just "bob@foo.com")
+blankUser :: User
+blankUser = User Nothing Nothing Nothing (fromList [])
+
 $(makeLenses ''User)
 
 instance FromJSON User where
-  parseJSON (Object v) = do
+  parseJSON (Object v) =
     User <$> v .:? "name"
     <*> v .:? "email"
     <*> v .:? "user_id"
     <*> v .: "custom_attributes"
   parseJSON _ = mzero
 
+instance ToJSON User where
+  toJSON (User n e u c) = object $ nonEmptys attributes
+    where nonEmptys xs = catMaybes $ fmap maybePair xs
```

Comment:
By default, a missing email will be rendered as `"email": null`. To avoid this we map the k/vs to Maybes then use `catMaybes` to rid the `Nothing`s